db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/pre_deployment.rb

```diff
@@ -1,93 +1,70 @@
-      elsif release.include?("Red Hat Enterprise Linux") &&
-            !$evm.root['automation_task'].automation_request.options[:attrs][:containerized]
-        commands = handle_rhel_subscriptions(commands)
-      end
-      commands.each do |cmd|
-        result = ssh_exec!(ssh, cmd)
-        unless result[:exit_code] == 0
-          $evm.root['automation_task'].message = "FAILED: couldn't execute command #{cmd}. ERROR: #{result[:stderr]}"
-          $evm.root['ae_result'] = "error"
-          failed_execute = true
-        end
-        break if failed_execute
-      end
-      unless failed_execute
-        $evm.root['ae_result'] = "ok"
-        $evm.root['automation_task'].message = "successful pre-deployment"
-      end
-    end
-  rescue
-    $evm.root['ae_result'] = "error"
-    $evm.root['automation_task'].message = "Cannot connect to deployment master " \
-                                           "(#{$evm.root['deployment_master']}) via ssh"
-  ensure
-    $evm.log(:info, "State: #{$evm.root['ae_state']} | Result: #{$evm.root['ae_result']} "\
-             "| Message: #{$evm.root['automation_task'].message}")
+  $evm.root['container_deployment'].perform_scp($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), "inventory.yaml", "inventory.yaml")
+  release = $evm.root['container_deployment'].perform_agent_commands($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), ["sudo cat /etc/redhat-release"])[:stdout]
+  if release.include?("CentOS")
+    commands.unshift("sudo yum install epel-release -y",
+                     "sudo curl -o /etc/yum.repos.d/maxamillion-origin-next-epel-7.repo #{REPO_URL}")
+  elsif release.include?("Red Hat Enterprise Linux") &&
+        !$evm.root['automation_task'].automation_request.options[:attrs][:containerized]
+    commands = handle_rhel_subscriptions(commands)
+    $evm.root['rhsub_user'], $evm.root['rhsub_pass'] = $evm.root['container_deployment'].rhsm_creds
+    $evm.root['rhsub_pool'] = $evm.root['automation_task'].automation_request.options[:attrs][:rhsub_sku]
+
+    $evm.root['container_deployment'].perform_agent_commands($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), ["sudo subscription-manager register --username='#{$evm.root['rhsub_user']}'  --password='#{$evm.root['rhsub_pass']}'"])
+    pool_id = $evm.root['container_deployment'].perform_agent_commands($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), ["sudo subscription-manager list --available --matches=#{$evm.root['rhsub_pool']} --pool-only"])[:stdout].split("\n").first.delete("\r")
+    $evm.root['container_deployment'].perform_agent_commands($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), ["sudo subscription-manager attach --pool=#{pool_id}"])
+  end
+  $evm.root['container_deployment'].perform_agent_commands($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), commands)
```

Comment:
These `.perform_agent_commands` are all same except for actual command(s), no?
Can you factor them into a helper function?
And/or assign `$evm.root['container_deployment']`, `$evm.root['deployment_master']`, `$evm.get_state_var(:ssh_user)` to local vars.